core/src/main/java/org/elasticsearch/discovery/zen/fd/NodesFaultDetection.java

```diff
@@ -145,14 +146,18 @@ protected void handleTransportDisconnect(DiscoveryNode node) {
     }
 
     private void notifyNodeFailure(final DiscoveryNode node, final String reason) {
-        threadPool.generic().execute(new Runnable() {
-            @Override
-            public void run() {
-                for (Listener listener : listeners) {
-                    listener.onNodeFailure(node, reason);
+        try {
+            threadPool.generic().execute(new Runnable() {
+                @Override
+                public void run() {
+                    for (Listener listener : listeners) {
+                        listener.onNodeFailure(node, reason);
+                    }
                 }
-            }
-        });
+            });
+        } catch (EsRejectedExecutionException ex) {
+            logger.debug("Rejected execution on {}", ex, reason);
```

Comment:
I chose `debug` as it's the same in `TransportService.raiseNodeDisconnected()`:
```
} catch (EsRejectedExecutionException ex) {
    logger.debug("Rejected execution on NodeDisconnected", ex);
}
```

Also I chose not to write "shutting down", because this does not always hold (EsRejectedExecutionException can be thrown in other cases).